### ISSUE TITLE
Keystore was tampered with, or password was incorrect

### ISSUE THREAD

**Issue by sgalkina** [Khezen/docker-logstash#4]
First of all, thank you for the images, it is really helpful.
I am getting an errors with the clean installation I can't debug though. If you have any idea about this, it would be amazing. 
In the README the the docker compose file for the whole ELK has volumes like this
```
        volumes:
            - /etc/logstash:/etc/logstash/conf.d
```
But when I try to deploy such thing, there is an error that the certificates are not found 
```# File does not exist or cannot be opened /etc/elasticsearch/searchguard/ssl/truststore.jks```
So assuming it needs the access to the directory with the certificates, I give it access to the elasticsearch volume with read-only permissions.
```
        volumes:
            - /etc/logstash:/etc/logstash/conf.d
            - /etc/elasticsearch:/etc/elasticsearch:ro
```
But after that there is an error
```
[[main]-pipeline-manager] ERROR logstash.agent - Pipeline aborted due to error {:exception=>java.io.IOException: Keystore was tampered with, or password was incorrect, :backtrace=>...
```
Environmental variables are all the same as in the initial configuration.
Thank you again.

**Comment by khezen** [comment]
Hi,

Could you give me more details about your environment(OS etc)? I am wondering if this could be the same as [khezen/docker-elasticsearch: issue 15](https://github.com/Khezen/docker-elasticsearch/issues/15).

**Comment by sgalkina** [comment]
It runs with Docker Cloud on Ubuntu 14.04. The paths are a bit different but it doesn't look like it makes a difference 
For Elasticsearch service:
```
  volumes:
    - '/usr/share/elasticsearch/data:/usr/share/elasticsearch/data'
    - '/usr/share/etc/elasticsearch:/usr/share/elasticsearch/config'
```
For Logstash service
```
  volumes:
    - '/usr/share/etc/logstash:/etc/logstash'
    - '/usr/share/etc/elasticsearch:/etc/elasticsearch:ro'
```
The issue you're referring to was also very helpful with local OSx setup

**Comment by sgalkina** [comment]
I think I found the problem! The `edit_config` script is supposed to change the configuration in a way that the password is now `LOGSTASH_PWD` https://github.com/Khezen/docker-logstash/blob/master/src/miscellaneous/edit_config.sh#L3
But `truststore_password`from config file https://github.com/Khezen/docker-logstash/blob/master/conf.d/logstash.conf#L39 is getting changed too with this regex, so it is now `LOGSTASH_PWD` instead of `TS_PWD`, thus the "incorrect password" error is raised
Also I must notice that after fixing that there were errors about `logstash` user not having permissions for an index. I got a temporal fix with setting `logstash` user to `INDEX_ALL` instead of `INDEX_OWNER`, but didn't look further into it.
Hope it helps somebody else, and thank you again for the effort on these images.
